### PR TITLE
[new layer] nav-flash

### PR DESCRIPTION
**Pull request by thanhvg**
Ported from Doom

### CODE REVIEW

File: layers/+misc/nav-flash/config.el [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(defvar nav-flash-exclude-commands
```
Comment (lebensterben) [inline layers/+misc/nav-flash/config.el:1]: Need to add header
https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org#file-header

File: layers/+misc/nav-flash/funcs.el [hunk 2]
```diff
@@ -0,0 +1,34 @@
+;;; funcs -*- lexical-binding: t; -*-
```
Comment (lebensterben) [inline layers/+misc/nav-flash/funcs.el:1]: Need to add header
https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org#file-header

File: layers/+misc/nav-flash/packages.el [hunk 3]
```diff
@@ -0,0 +1,70 @@
+;;; packages.el --- nav-flash layer packages file for Spacemacs.
```
Comment (lebensterben) [inline layers/+misc/nav-flash/packages.el:1]: Need to update header
https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org#file-header

**Review by lebensterben — CHANGES_REQUESTED** [review]

File: layers/+misc/nav-flash/packages.el [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  ;; persp and eyebrowse
+  (advice-add 'persp-switch :after #'nav-flash/delayed-blink-cursor-h)
+
+  :config
+  ;; emacs 27 extend face
+  (when (fboundp 'set-face-extend)
+    (with-eval-after-load "nav-flash"
+      (set-face-extend 'nav-flash-face t))))
+;;; packages.el ends here
```
Comment (smile13241324) [inline layers/+misc/nav-flash/packages.el:70]: No packages.el ends here line please

File: layers/+misc/nav-flash/packages.el [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+;; Author: Thanh Vuong <thanhvg@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;;; Code:
```
Comment (smile13241324) [inline layers/+misc/nav-flash/packages.el:14]: Please remove these two comments

**Review by smile13241324 — CHANGES_REQUESTED** [review]
New functionality is ok, still you should adress the points lebensterben pointed out above.

**Review by smile13241324 — APPROVED** [review]
Looks good now

**Comment by lebensterben** [comment]
This PR has a major issue.

It's ported from Doom Emacs, which has MIT license.
But the files added in this PR wrongly used GPL-v3 license.

I'm working on a fix now.

**Comment by thanhvg** [comment]
@lebensterben  you seem to have a lot of free time on this frivolous matter that imo is truly harmful to the community. Google told me MIT is compatible with GPL so I don't see any problem here. 

But I'm asking you @syl20bnr @smile13241324 and @JAremko those who I am assumed to be the owners of this project to be clear on this matter. Are my contributions breaking any rule? Do you allow code from SO, reddit, blogs or other elisp projects. If you don't  then please consider removing all of my changes because most of them I got inspired by the sources mentioned above or they were plain copy pasta.

Thanks.

**Comment by lebensterben** [comment]
@thanhvg 

GPL 3.0 is compatible with MIT license. But you must include the copyright notice of MIT even after you re-licensed the work.

MIT license requires:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

A relative issue is that you didn't mention the copyright of Henrik Lissner.

Even though there are additional changes made upon his original work, his copyright persists.

I've already fixed this issue in a this commit c2903ad64a6317d0b88a3ce721bebe5aa1c14896.

---

>  Are my contributions breaking any rule? Do you allow code from SO, reddit, blogs or other elisp projects. If you don't then please consider removing all of my changes because most of them I got inspired by the sources mentioned above or they were plain copy pasta.

https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org#license reads:

> For files not belonging to Spacemacs like local packages and libraries, refer to the header file. Those files should not have an empty header, we may not accept code without a proper header file.

You must make contributions with due diligence. If your PR contains or is based on on other people's non-trivial work, you must respect the copyright of the original author, as well as the original license, if there is.

**Comment by JAremko** [comment]
@thanhvg   

The owner is  @syl20bnr you can try contacting him via email. I don't think he has GitHub mention notifications on.

**Pull request merged**